File: engine/engine-model/src/main/scala/tools/packet/inventory/SlotLimitUpdate.scala
package tools.packet.inventory

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class SlotLimitUpdate(private var _inventoryType: Int, private var _newLimit: Int) extends PacketInput {
  def inventoryType: Int = _inventoryType

  def newLimit: Int = _newLimit

  override def opcode(): SendOpcode = SendOpcode.INVENTORY_GROW
}

File: engine/engine-model/src/main/java/tools/packet/inventory/SlotLimitUpdate.java
package tools.packet.inventory;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record SlotLimitUpdate(Integer inventoryType, Integer newLimit) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.INVENTORY_GROW;
   }
}